compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -131,58 +124,70 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
     // See #74595 for more details about this.
     let concrete = infcx.const_eval_resolve(param_env, uv.expand(), Some(span));
 
-    if concrete.is_ok() && uv.substs.has_param_types_or_consts() {
-        match infcx.tcx.def_kind(uv.def.did) {
-            DefKind::AnonConst | DefKind::InlineConst => {
-                let mir_body = infcx.tcx.mir_for_ctfe_opt_const_arg(uv.def);
-
-                if mir_body.is_polymorphic {
-                    future_compat_lint();
-                }
-            }
-            _ => future_compat_lint(),
-        }
-    }
-
-    // If we're evaluating a foreign constant, under a nightly compiler without generic
-    // const exprs, AND it would've passed if that expression had been evaluated with
-    // generic const exprs, then suggest using generic const exprs.
-    if concrete.is_err()
-        && tcx.sess.is_nightly_build()
+    match concrete {
+      // If we're evaluating a foreign constant, under a nightly compiler without generic
+      // const exprs, AND it would've passed if that expression had been evaluated with
+      // generic const exprs, then suggest using generic const exprs.
+      Err(_) if tcx.sess.is_nightly_build()
         && !uv.def.did.is_local()
```

Comment:
i realized that this check isǹ't necessary as its implied by `let Ok(Some(ct)) = AbstractConst::new(tcx, uv)`
```suggestion
```